specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/examples/BastionGetSessionRecordingSasUrl.json

```diff
@@ -0,0 +1,18 @@
+{
+  "parameters": {
+    "api-version": "2024-01-01",
+    "subscriptionId": "subid",
+    "resourceGroupName": "rg1",
+    "bastionHostName": "bastionhosttenant"
+  },
+  "responses": {
+    "200": {
+      "body": "https://storageAccount.blob.core.windows.net/container?sp=rcwl&st=1970-01-01T00:00:00Z&se=2099-12-31T23:59:59Z&spr=https&sv=2022-11-02&sr=c"
```

Comment:
I'd like to strongly recommend doing this API a more future-proof way, because I've been bitten by using designs like this in the past.
Instead of returning a body which is a _raw string_, I'd recommend this API instead return a very simple JSON payload (with whatever property name makes sense to you for this) e.g.

```
{
    "sasUriTemplate": "https://storageAccount.blob.core.windows.net/container?sp=rcwl&st=1970-01-01T00:00:00Z&se=2099-12-31T23:59:59Z&spr=https&sv=2022-11-02&sr=c"
}

Reply:
This will only be used to look for storage account and container names so keeping it like this for now

Reply:
I agree that using a string payload is not a good idea. Note: the RPC requires a JSON payload: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests)